homeassistant/components/tplink/fan.py

```diff
@@ -0,0 +1,105 @@
+"""Support for TPLink Fan devices."""
+
+import logging
+import math
+from typing import Any
+
+from kasa import Device, Module
+from kasa.interfaces import Fan
+
+from homeassistant.components.fan import FanEntity, FanEntityFeature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+from homeassistant.util.scaling import int_states_in_range
+
+from . import TPLinkConfigEntry
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, async_refresh_after
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up fans."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    device = parent_coordinator.device
+    entities: list[CoordinatedTPLinkEntity] = []
+    if Module.Fan in device.modules:
+        entities.append(
+            TPLinkFan(device, parent_coordinator, device.modules[Module.Fan])
+        )
+    entities.extend(
+        TPLinkFan(child, parent_coordinator, child.modules[Module.Fan], parent=device)
+        for child in device.children
+        if Module.Fan in child.modules
+    )
+    async_add_entities(entities)
+
+
+SPEED_RANGE = (1, 4)  # off is not included
+
+
+class TPLinkFan(CoordinatedTPLinkEntity, FanEntity):
+    """Representation of a fan for a TPLink Fan device."""
+
+    _attr_speed_count = int_states_in_range(SPEED_RANGE)
+    _attr_supported_features = FanEntityFeature.SET_SPEED
+
+    def __init__(
+        self,
+        device: Device,
+        coordinator: TPLinkDataUpdateCoordinator,
+        fan_module: Fan,
```

Comment:
```suggestion
        coordinator: TPLinkDataUpdateCoordinator,
        *,
        fan_module: Fan,
```
Makes it consistent with the base class which accepts only device & coordinator as positional args.

Reply:
That's done